src/ui/osx/TogglDesktop/Feature/Permission/SystemPermissionManager.swift

```diff
@@ -0,0 +1,169 @@
+        alert.alertStyle = .warning
+        alert.addButton(withTitle: "Open")
+        alert.addButton(withTitle: "Later")
+
+        let result = alert.runModal()
+        if result == .alertFirstButtonReturn {
+            complete()
+        }
+    }
+
+    private func isAlreadyRequestSystemPermission(_ permission: Permission) -> Bool {
+        switch permission {
+        case .screenRecording:
+            return UserDefaults.standard.bool(forKey: Constants.ScreenRecordingPermissionGranted)
+        }
+    }
+
+    private func tryOpeningSystemPreference(for permission: Permission) {
+
+        // Open the Screen Recording tab
+        // https://stackoverflow.com/a/48139877/3127477
+        if let url = URL(string: permission.urlScheme) {
+            NSWorkspace.shared.open(url)
+            return
+        }
+
+        // Old-fashioned way
+        // https://github.com/cho45/KeyCast/blob/master/KeyCast/Accessibility.swift
+        let url = URL(fileURLWithPath: "/System/Library/PreferencePanes/Security.prefPane")
+        NSWorkspace.shared.open(url)
+    }
+}
+
+// MARK: Screen Recording
+
+extension SystemPermissionManager {
+
+    // CGWindowListCopyWindowInfo is not reliable because some windows has the name even the permission is OFF
+    // Reliable way to check whether the permission is enable or not
+    // However it would trigger the system alert too
+    private func canRecordScreen() -> Bool {
+        let mainDisplay    = CGMainDisplayID()
+        let displayBounds  = CGDisplayBounds(mainDisplay)
+        let recordingQueue = DispatchQueue.global(qos: .background)
+        let displayStreamProps : [CFString : Any] = [
+            CGDisplayStream.preserveAspectRatio: kCFBooleanTrue,
+            CGDisplayStream.showCursor:          kCFBooleanTrue,
+            CGDisplayStream.minimumFrameTime:    60,
+        ]
+
+        let displayStream = CGDisplayStream(
+            dispatchQueueDisplay: mainDisplay,
+            outputWidth:          Int(displayBounds.width),
+            outputHeight:         Int(displayBounds.height),
+            pixelFormat:          Int32(kCVPixelFormatType_32BGRA),
+            properties:           displayStreamProps as CFDictionary,
+            queue:                recordingQueue,
+            handler: { status, displayTime, frameSurface, updateRef in
+                print("is only called once")
+        }
+        )
+        return displayStream != nil
+    }
+
+    private func triggerScreenRecordingPermissionAlert() {
+        guard let window = CoreGraphicsApis.windows()?.last else { return }
+        let windowNumber = (window[kCGWindowNumber as String] as? CGWindowID) ?? UInt32(0)
+
+        // There is no func to explicit trigger the permission alert
+        // Have to execute CGWindowListCreateImage -> OS will present the alert once
+        CoreGraphicsApis.image(windowNumber)
```

Comment:
There is no official APIs to trigger the System Permission Alert -> We have to implicitly trigger from `CGWindowListCreateImage`